vpr/src/route/rr_graph_storage.h

```diff
@@ -584,6 +617,14 @@ class t_rr_graph_storage {
     vtr::vector<RREdgeId, RRNodeId> edge_dest_node_;
     vtr::vector<RREdgeId, short> edge_switch_;
 
+    // Sets of non-configurably connected nodes
```

Comment:
Need a comment on what these non-configurable edge sets are.  May go here, or at top of rr_graph_storage (with other top-level rr_graph comments).
Non-configurable edges are switches in the rr-graph that cannot be turned off. They may be metal shorts, or non-tri-stateable buffers: in both cases the switch always connects two nodes, and cannot be turned off no matter what the router does.
These non-configurable edges are useful for describing L and T-shaped wires as the rr_graph can only store X- or Y- directed (straight) wires.  Using non-configurable edges lets us model metal shorts to describe the L or T-shaped wire. Non-configurable edges can also occur with non-tristatable buffers, which can be used in clock networks or on wires so long that they have internal buffering that can't be turned off.
If the router uses one member of a non-configurable node set (i.e. rr_nodes connected with non-configurable edges) then it has implicitly used them all, as the others are directly wired to the one used.

... Keith: add explanation on exactly what is in non_config_set_nodes_ and non_config_first_node_idx_

Reply:
Done